woocommerce/woocommerce-android
# Tracking is done on the main thread

Closes: #11292


### Description

While working on another issue, I noticed that in many cases, we call the "track" method on the main thread. Inside this, we hit the device's IO. This can make the app less responsive and jerky. The PR wraps this code in a courtines with `appCoroutineScope` scope. As this potentially dangerous change, @hichamboushaba @JorgeMucientes, would you mind thinking of potential issues with this approach?

### Testing instructions

* Use the app
* Notice that tracking of the events works (live view can be used)
